src/SkirtBrim.cpp

```diff
@@ -387,15 +387,22 @@ Polygons SkirtBrim::getFirstLayerOutline(const int extruder_nr /* = -1 */)
         const bool skirt_around_prime_tower_brim = storage.primeTower.enabled && global_settings.get<bool>("prime_tower_brim_enable");
         const bool include_prime_tower = ! skirt_around_prime_tower_brim; // include manually otherwise
 
-        const int skirt_height = global_settings.get<int>("skirt_height");
         first_layer_outline = Polygons();
-        for (int i_layer = layer_nr; i_layer <= skirt_height; ++i_layer)
+        int skirt_height = 0;
+        for (const auto& extruder : Application::getInstance().current_slice->scene.extruders)
         {
-            first_layer_outline =
-                first_layer_outline.unionPolygons
-                (
-                    storage.getLayerOutlines(i_layer, include_support, include_prime_tower, external_only, extruder_nr)
-                );
+            skirt_height = std::min(std::max(skirt_height, extruder.settings.get<int>("skirt_height")), static_cast<int>(storage.print_layer_count));
+        }
+        for (int i_layer = layer_nr; i_layer < skirt_height; ++i_layer)
+        {
+            for (const auto& extruder : Application::getInstance().current_slice->scene.extruders)
+            {
+                first_layer_outline =
+                    first_layer_outline.unionPolygons
+                    (
+                        storage.getLayerOutlines(i_layer, include_support, include_prime_tower, external_only, extruder.extruder_nr)
+                    );
+            }
         }
```

Comment:
I'm not sure if this is needed; we have an `extruder_nr` parameter, the documentation for this parameter reads
```
     * \param extruder_nr The extruder for which to get the outlines. -1 to include outliens for all extruders
```
So we should only include the layer_outline for all extruders if `extruder_nr` is set to `-1`. When executing `storage.getLayerOutlines` with this `-1` argument should already return the outline of all extruders.